pkg/cmd/run/shared/artifacts.go

```diff
@@ -17,37 +17,70 @@ type Artifact struct {
 	Expired     bool   `json:"expired"`
 }
 
+type artifactsPayload struct {
+        Artifacts []Artifact
+}
+
 func ListArtifacts(httpClient *http.Client, repo ghrepo.Interface, runID string) ([]Artifact, error) {
-	perPage := 100
-	path := fmt.Sprintf("repos/%s/%s/actions/artifacts?per_page=%d", repo.RepoOwner(), repo.RepoName(), perPage)
-	if runID != "" {
-		path = fmt.Sprintf("repos/%s/%s/actions/runs/%s/artifacts?per_page=%d", repo.RepoOwner(), repo.RepoName(), runID, perPage)
-	}
+	var results []Artifact
 
-	req, err := http.NewRequest("GET", ghinstance.RESTPrefix(repo.RepoHost())+path, nil)
-	if err != nil {
-		return nil, err
-	}
+	path := fmt.Sprintf("repos/%s/%s/actions/", repo.RepoOwner(), repo.RepoName())
+	if runID == "" {
+		path += "artifacts"
+        } else {
+		path += fmt.Sprintf("runs/%s/artifacts", runID)
+        }
 
-	resp, err := httpClient.Do(req)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
+	url := fmt.Sprintf("%s%s?per_page=100", ghinstance.RESTPrefix(repo.RepoHost()), path)
 
-	if resp.StatusCode > 299 {
-		return nil, api.HandleHTTPError(resp)
-	}
+	for {
+		var payload artifactsPayload
+		nextURL, err := apiGet(httpClient, url, &payload)
+		if err != nil {
+			return nil, err
+		}
+		results = append(results, payload.Artifacts...)
 
-	var response struct {
-		TotalCount uint16 `json:"total_count"`
-		Artifacts  []Artifact
+		if nextURL == "" {
+			break
+		}
+		url = nextURL
 	}
 
-	dec := json.NewDecoder(resp.Body)
-	if err := dec.Decode(&response); err != nil {
-		return response.Artifacts, fmt.Errorf("error parsing JSON: %w", err)
-	}
+	return results, nil
+}
+
+func apiGet(httpClient *http.Client, url string, data interface{}) (string, error) {
+	req, err := http.NewRequest("GET", url, nil)
+        if err != nil {
+                return "", err
+        }
+
+        resp, err := httpClient.Do(req)
+        if err != nil {
+                return "", err
+        }
+        defer resp.Body.Close()
+
+        if resp.StatusCode > 299 {
+                return "", api.HandleHTTPError(resp)
+        }
+
+        dec := json.NewDecoder(resp.Body)
+        if err := dec.Decode(data); err != nil {
+                return "", err
+        }
+
+        return findNextPage(resp), nil
+}
+
+var linkRE = regexp.MustCompile(`<([^>]+)>;\s*rel="([^"]+)"`)
 
-	return response.Artifacts, nil
+func findNextPage(resp *http.Response) string {
+        for _, m := range linkRE.FindAllStringSubmatch(resp.Header.Get("Link"), -1) {
+                if len(m) > 2 && m[2] == "next" {
+                        return m[1]
+                }
+        }
+        return ""
 }
```

Comment:
Btw, I copied [this code from elsewhere in the repository](https://github.com/cli/cli/blob/trunk/pkg/cmd/api/pagination.go#L13-L22), and I would like to be able to reuse that instead of copying. I wasn't able to get import/referencing to work; I would try:
```go
import (
...
  cmdApi "github.com/cli/cli/v2/pkg/cmd/api"
)
...
cmdApi.findNextPage()
```
but that would fail with `findNextPage() is undefined` in `api` package...